### ISSUE TITLE
Intermittently SafeAreaView not working with in-call extra height on iOS

### ISSUE THREAD

**Issue by townmulti** [react-native-community/react-native-safe-area-view#31]
When I'm already in a call and then open an app using expo, sometimes SafeAreaView doesn't apply the extra height needed. It does the same with or without using the package expo/status-bar-height.  I'm using react-navigation (version 1.5.11) SafeAreaView and here is a screenshot of it in the expo app using version 27.0.1.

![image](https://user-images.githubusercontent.com/10806345/40064475-362a13ac-5825-11e8-8460-eb405d483f57.png)

**Comment by mightyaleksey** [comment]
Since safe-area-view can't detect whether `StatusBar` component is `translucent` or not on android, you can set the statusBarHeight manually via `setStatusBarHeight` static method.

```js
import {Platform} from 'react-native';
import SafeAreaView from 'react-native-safe-area-view';

if (Platform.OS === 'android') {
  SafeAreaView.setStatusBarHeight(25);
}
```

or if you use `react-navigation`, then:

```js
import {Platform} from 'react-native';
import {SafeAreaView} from 'react-navigation';

if (Platform.OS === 'android') {
  SafeAreaView.setStatusBarHeight(25);
}
```

**Comment by slorber** [comment]
@sullenor it looks to me that `setStatusBarHeight` does not trigger a re-render, I think it would be helpful to make sure all mounted instances re-render when this setter is called

**Comment by mightyaleksey** [comment]
@slorber yeah, it is. Currently it will work in case you set it before initial rendering.

**Comment by slorber** [comment]
i've made a pr for that https://github.com/react-community/react-native-safe-area-view/pull/38

**Comment by brentvatne** [comment]
this should be resolved in version 1.0 which was released today, thanks for reporting!
